transport/tcp/tcp.go

```diff
@@ -160,15 +168,12 @@ func (t *tcpTran) Scheme() string {
 
 func (t *tcpTran) NewDialer(addr string, sock mangos.Socket) (mangos.PipeDialer, error) {
 	var err error
-	d := &dialer{sock: sock, opts: newOptions()}
-
 	if addr, err = mangos.StripScheme(t, addr); err != nil {
 		return nil, err
 	}
 
-	if d.addr, err = mangos.ResolveTCPAddr(addr); err != nil {
```

Comment:
I think we should still resolve this at dialer creation.  I'm fine with resolving it *again*, at dial time.  The reason to do this is that we want creation of the dialer to fail if it is given a bad name.  Otherwise, deferring it to dialing becomes much harder to debug later in production applications.